CRM/Contact/BAO/ProximityQuery.php

```diff
@@ -333,15 +333,12 @@ public static function process(&$query, &$values) {
       )
     );
 
-    $fnName = isset($config->geocodeMethod) ? $config->geocodeMethod : NULL;
-    if (empty($fnName)) {
-      CRM_Core_Error::fatal(ts('Proximity searching requires you to set a valid geocoding provider'));
-    }
-
     $query->_tables['civicrm_address'] = $query->_whereTables['civicrm_address'] = 1;
 
-    require_once str_replace('_', DIRECTORY_SEPARATOR, $fnName) . '.php';
-    $fnName::format($proximityAddress);
+    if (!CRM_Core_BAO_Address::addGeocoderData($proximityAddress)) {
+      throw new CRM_Core_Exception(ts('Proximity searching requires you to set a valid geocoding provider'));
```

Comment:
@eileenmcnaughton i feel like the `ts()` shouldn't be here

Reply:
I feel like we discussed that on an earlier version & agreed it should - because the thing insided ts() is sent to the translators - heaps of cases if you grep - e.g relationshipTYpe

```
  public static function del($relationshipTypeId) {
    // make sure relationshipTypeId is an integer
    // @todo review this as most delete functions rely on the api & form layer for this
    // or do a find first & throw error if no find
    if (!CRM_Utils_Rule::positiveInteger($relationshipTypeId)) {
      throw new CRM_Core_Exception(ts('Invalid relationship type'));
    }
```

Reply:
(@mlutfy ...

Reply:
but on the other hand, i see plenty of examples where CRM_Core_Exception is used without ts and i guess it seems odd that its in some but not others and won't this mean that the exception message the end user gets won't necessarily be found in the code base making debugging hard?

Reply:
Hmm - let's see what @mlufty says & I will submit a follow up patch if he says it should be changed

Reply:
We don't have a specific rule for this, it's arbitrary depending on who is likely to see the error, and how often. Seems OK either way. Probably better with `ts()` in this case.

Reply:
gotta love arbitrary :-) @ErikHommel & I always used to have an opinion to delegate to :-)